src/etc/inc/captiveportal.inc

```diff
@@ -2841,4 +2899,24 @@ function captiveportal_isip_logged($clientip) {
 		return $cpentry;
 	}
 }
+
+// Return true if the CARP status of at least one interface of a captive portal zone is in backup mode
+function captiveportal_ha_is_node_in_backup_mode() {
+	global $config, $cpzone;
+
+	$cpinterfaces = explode(",", $config['captiveportal'][$cpzone]['interface']);
+
+	if (is_array($config['virtualip']['vip'])) {
+		foreach ($cpinterfaces as $interface) {
+			foreach ($config['virtualip']['vip'] as $vip) {
```

Comment:
This foreach and its related code seems like a good candidate to move to a more general function in `pfsense-utils.inc`, where the other functions like `get_carp_interface_status()` resides. I could see a function which takes a pfSense interface (like LAN) and returns a MASTER/BACKUP type value being useful.

Then this function would only need (a) check if CARP is enabled, and (b) loop through the CP interfaces calling that function.